### PR TITLE
Adding ingress fan-out and host sub-domain generation

### PR DESCRIPTION
**Pull request by seshapad**
Signed-off-by: seshapad <seshapad@in.ibm.com>

### CODE REVIEW

**Review by ashokponkumar — CHANGES_REQUESTED** [review]

File: internal/types/ir.go [hunk 1]
```diff
@@ -44,19 +44,22 @@ type IR struct {
...
-	Replicas      int
-	Networks      []string
-	ExposeService bool
-	Daemon        bool //Gets converted to DaemonSet
+	Name        string
+	Annotations map[string]string
+	Labels      map[string]string
+	Replicas    int
+	Networks    []string
+	IngressPath string //Ingress fan-out path
```
[the remark this answers is not in the document]
Comment (seshapad) [reply internal/types/ir.go:61]: @ashokponkumar 
Are you suggesting that we provide serviceRelPath for each service in the path file? That would be nice actually...will allow users to customize.

File: internal/types/ir.go [hunk 2]
```diff
@@ -44,19 +44,22 @@ type IR struct {
 	CachedObjects     []runtime.Object
 
 	Values outputtypes.HelmValues
+
+	IngressHost    string
```
[the remark this answers is not in the document]
Comment (seshapad) [reply internal/types/ir.go:48]: Yes. This should be queried from the cluster.

File: internal/types/ir.go [hunk 3]
```diff
@@ -152,6 +155,11 @@ func (c *Container) Merge(newc Container) bool {
 	return false
 }
 
+// IsServiceExposed checks if a service is exposed through ingress or not
+func (service *Service) IsServiceExposed() bool {
```
[the remark this answers is not in the document]
Comment (seshapad) [reply internal/types/ir.go:159]: Could you elaborate?

File: internal/parameterizer/ingressparameterizer.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	irtypes "github.com/konveyor/move2kube/internal/types"
+	plantypes "github.com/konveyor/move2kube/types/plan"
+)
+
+// ingressParameterizer parameterizes the image names
+type ingressParameterizer struct {
+}
+
+func (ip ingressParameterizer) parameterize(ir *irtypes.IR) error {
+	isHelm := (ir.Kubernetes.ArtifactType != plantypes.Yamls)
```
[the remark this answers is not in the document]
Comment (seshapad) [reply internal/parameterizer/ingressparameterizer.go:29]: ok.

File: internal/types/ir.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (seshapad) [reply internal/types/ir.go:159]: empty path means we are not exposing the service.

File: internal/types/ir.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (HarikrishnanBalagopal) [reply internal/types/ir.go:61]: @ashokponkumar @seshapad so can we talk about that? it's related to https://github.com/konveyor/move2kube/issues/61

**Review by ashokponkumar — APPROVED** [review]

**Pull request merged**